app/src/main/java/io/intrepid/contest/screens/contestcreation/namecontest/NameContestFragment.java

```diff
@@ -25,24 +23,42 @@
     ImageView trophyIcon;
 
     @Override
-    public void onCreate(@Nullable Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-    }
-
-    @Override
     protected int getLayoutResourceId() {
         return R.layout.fragment_edit_contest_name;
     }
 
     @OnTextChanged(R.id.hint_label_edit_text)
     public final void onTextChanged(CharSequence newName) {
-        if(TextUtils.isEmpty(newName)){
-            trophyIcon.setVisibility(VISIBLE);
-        }else{
-            trophyIcon.setVisibility(GONE);
+        if (TextUtils.isEmpty(newName)) {
+            onNextInvalidated();
+        } else {
+            onNextValidated();
         }
     }
 
+    public void onNextInvalidated() {
+        presenter.onNextInvalidated();
+    }
+
+    @Override
+    public void onNextValidated() {
+        presenter.onNextValidated();
+    }
+
+    @Override
+    public void enableNext() {
+        trophyIcon.setVisibility(GONE);
+        EditContestContract activity = (EditContestContract) getActivity();
```

Comment:
It seems to me that this activity could be a class field initialized in `onViewCreated`